### ISSUE TITLE
Chunking ignores topicheads

### ISSUE THREAD

**Issue by stokecn** [dita-ot/dita-ot#2428]
Using DITA-OT 2.3. Similar to [https://github.com/dita-ot/dita-ot/issues/686](https://github.com/dita-ot/dita-ot/issues/686) I think, but that issue is closed since 2013.

When chunk="to-content" is set on an ancestor (in my case) the result chunk does not include generated content for a descendant topichead. Since PDF output does support topichead, this results in a topichead generated out of order.

The chunk attribute in my scenario is desired for another transtype but the map is shared and also used in PDF docs. Both transtypes share this issue but the PDF case is the easiest to demonstrate.

[topichead_behavior.zip](https://github.com/dita-ot/dita-ot/files/362445/topichead_behavior.zip)

**Comment by drmacro** [comment]
In addition, if there are two levels of topichead, both of which specify chunk="to-content select-topic", the child topichead does not result in a chunk and any topicref children of the child topichead are treated as though they had been select-branch chunked into the parent topichead's chunk (even though the parent topichead's chunk is correctly created as a single topic chunk reflecting only the topichead's navigation title).

My test case is here: https://github.com/dita-community/dita-test-cases/tree/master/topichead-chunking

It exercises each possible combination of chunk values in a two-level topichead situation.

I get the same incorrect result with 1.8.5. With 2.3 I get an NPE failure in the chunk processing:

```
/Users/ekimber/dita-ot/dita-ot-2.3/plugins/org.dita.base/build_preprocess.xml:313: java.lang.NullPointerException
	at org.dita.dost.writer.ChunkTopicParser.processChunk(ChunkTopicParser.java:262)
	at org.dita.dost.writer.ChunkTopicParser.write(ChunkTopicParser.java:63)
	at org.dita.dost.reader.ChunkMapReader.processChunk(ChunkMapReader.java:520)
	at org.dita.dost.reader.ChunkMapReader.processTopicref(ChunkMapReader.java:312)
	at org.dita.dost.reader.ChunkMapReader.processChildTopicref(ChunkMapReader.java:497)
	at org.dita.dost.reader.ChunkMapReader.processTopicref(ChunkMapReader.java:336)
	at org.dita.dost.reader.ChunkMapReader.process(ChunkMapReader.java:140)
	at org.dita.dost.writer.AbstractDomFilter.read(AbstractDomFilter.java:48)
	at org.dita.dost.reader.ChunkMapReader.read(ChunkMapReader.java:104)
	at org.dita.dost.module.ChunkModule.execute(ChunkModule.java:86)
	at org.dita.dost.pipeline.PipelineFacade.execute(PipelineFacade.java:70)
	at org.dita.dost.invoker.ExtensibleAntInvoker.execute(ExtensibleAntInvoker.java:232)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:441)
	at org.apache.tools.ant.taskdefs.CallTarget.execute(CallTarget.java:105)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
	at org.apache.tools.ant.helper.DefaultExecutor.executeTargets(DefaultExecutor.java:41)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
	at org.apache.tools.ant.Main.runBuild(Main.java:851)
	at org.apache.tools.ant.Main.startAnt(Main.java:235)
	at org.apache.tools.ant.launch.Launcher.run(Launcher.java:280)
	at org.apache.tools.ant.launch.Launcher.main(Launcher.java:109)
```

**Comment by jelovirt** [comment]
Been refactoring chunking code and accidentally bumped into this issue. Reproduced.

**Comment by raducoravu** [comment]
@jelovirt you seem to have fixed this problem so issue just needs to be closed.

**Comment by robander** [comment]
This issue is fixed for PDF with #3216.

There is still an open issue with at least some cases for HTML, as reported in #2787 using Eliot's samples from above.
